### PR TITLE
pubsub: add TestOpenTopic, TestOpenSubscription to awspubsub_test

### PR DESCRIPTION
**Pull request by ijt**
This makes progress on #1243.

### CODE REVIEW

**Comment by jba** [comment]
Don't we have conformance tests for these?

**Comment by ijt** [comment]
The conformance tests only cover the drivers, not the concrete constructors.

**Comment by jba** [comment]
Ah. LGTM then.

File: pubsub/awspubsub/awspubsub_test.go [hunk 1]
```diff
@@ -42,6 +42,36 @@ import (
 
 const region = "us-east-2"
 
+func TestOpenTopic(t *testing.T) {
+	ctx := context.Background()
+	dth, err := newHarness(ctx, t)
+	if err != nil {
+		t.Fatal(err)
+	}
+	h := dth.(*harness)
```
[the remark this answers is not in the document]
Comment (ijt) [reply pubsub/awspubsub/awspubsub_test.go]: I removed the harnesses. PTAL.

**Pull request merged**